core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -238,6 +251,13 @@ public String getId() {
     }
 
     /**
+     * The unique ephemeral id of the node.
```

Comment:
can we add some docs as to what we mean with ephemeral - most notably when it changes.